lib/commands/general.js

```diff
@@ -184,5 +186,23 @@ commands.setUrl = async function (url) {
   return await iosCommands.general.setUrl.call(this, url);
 };
 
+commands.getDevicePixelRatio = iosCommands.device.getDevicePixelRatio;
```

Comment:
I assume these two commands use uiAutomator primitives to get the desired values and won't work with XCTest

Reply:
the function `getDevicePixelRatio` itself does not make use of any UIAutomation primitives:

```js
commands.getDevicePixelRatio = async function () {
  const screenHeight = await this.getScreenHeight();
  // There is no way to grab scale from UIAutomation. All devices have scale 2.0 except iPhone 6+, iPhone 6s+, iPhone7+ - 3.0.
  const screenHeightOfPlusDevices = 736;
  return screenHeight === screenHeightOfPlusDevices ? 3.0 : 2.0;
};
```

since we override `getScreenHeight`, it will work.

Reply:
I'd rather get this value directly from WDA. XCTest has properties, which allow to get display scale factor

Reply:
Also, the current implementation might not work properly for iPads

Reply:
if we can get an accurate figure straight from WDA, even better. can you point me to where in the WDA code i can look to figure out how to make the query?

Reply:
yw: https://github.com/appium/WebDriverAgent/pull/20